File: Problems/844_Backspace_String_Compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        s_stack = []
        for c in s:
            if c == '#' and len(s_stack) != 0:
                s_stack.pop()
            elif c != '#':
                s_stack.append(c)
        
        t_stack = []
        for c in t:
            if c == '#' and len(t_stack) != 0:
                t_stack.pop()
            elif c != '#':
                t_stack.append(c)
        return s_stack == t_stack

def main():
    solution = Solution()
    s = "a##c"
    t = "#a#c"
    print(solution.backspaceCompare(s, t))

if __name__ == "__main__":
    main()

File: Problems/844_Backspace_String_Compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        stack<char> stackS, stackT;
        for(auto c: s){
            if(c=='#'){
                if(stackS.size())stackS.pop();
            }else{
                stackS.push(c);
            }
        }
        for(auto c: t){
            if(c=='#'){
                if(stackT.size())stackT.pop();
            }else{
                stackT.push(c);
            }
        }
        return stackS==stackT;
    }
};